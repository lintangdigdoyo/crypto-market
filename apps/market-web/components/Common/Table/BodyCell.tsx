import classNames from "classnames";

interface BodyCellProps extends React.TdHTMLAttributes<HTMLTableCellElement> {
  className?: string;
  children: React.ReactNode;
}

const BodyCell = ({ className, children, ...restProps }: BodyCellProps) => {
  return (
    <td
      className={classNames("p-5 border-t border-slate-200", className)}
      {...restProps}
    >
      {children}
    </td>
  );
};

export default BodyCell;
