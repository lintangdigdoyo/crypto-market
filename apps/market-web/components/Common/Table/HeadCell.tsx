import classNames from "classnames";

interface HeadCellProps extends React.ThHTMLAttributes<HTMLTableCellElement> {
  className?: string;
  children: React.ReactNode;
}

const HeadCell = ({ className, children, ...restProps }: HeadCellProps) => {
  return (
    <th
      className={classNames("p-5 text-gray font-semibold", className)}
      {...restProps}
    >
      {children}
    </th>
  );
};

export default HeadCell;
