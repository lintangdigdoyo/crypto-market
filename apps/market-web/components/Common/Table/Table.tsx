import classNames from "classnames";

interface TableProps extends React.TableHTMLAttributes<HTMLTableElement> {
  className?: string;
  children: React.ReactNode;
}

const Table = ({ className, children, ...restProps }: TableProps) => {
  return (
    <table
      className={classNames(
        "w-full border border-slate-200 border-separate border-spacing-0 rounded-lg",
        className
      )}
      {...restProps}
    >
      {children}
    </table>
  );
};

export default Table;
