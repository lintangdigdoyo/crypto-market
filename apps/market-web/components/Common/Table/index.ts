export { default as Table } from "./Table";
export { default as HeadCell } from "./HeadCell";
export { default as BodyCell } from "./BodyCell";
