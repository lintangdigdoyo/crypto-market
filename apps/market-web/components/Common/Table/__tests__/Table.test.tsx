import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import Table from "../Table";

describe("Table components", () => {
  it("should renders Table correctly", () => {
    const props = {
      children: <>children</>,
    };
    const component = renderer.create(<Table {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
