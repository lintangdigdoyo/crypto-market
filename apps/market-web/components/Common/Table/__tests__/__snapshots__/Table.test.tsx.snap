// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table components should renders Table correctly 1`] = `
<table
  className="w-full border border-slate-200 border-separate border-spacing-0 rounded-lg"
>
  children
</table>
`;
